crates/core/app/tests/common/mod.rs

```diff
@@ -89,3 +89,46 @@ impl TempStorageExt for TempStorage {
         self.apply_genesis(Default::default()).await
     }
 }
+
+#[async_trait]
+pub trait TestNodeExt: Sized {
+    async fn fast_forward_to_next_epoch(
+        &mut self,
+        storage: &TempStorage,
+    ) -> anyhow::Result<penumbra_sct::epoch::Epoch>;
+}
+
+#[async_trait]
+impl<C> TestNodeExt for TestNode<C>
+where
+    C: tower::Service<
+            tendermint::v0_37::abci::ConsensusRequest,
+            Response = tendermint::v0_37::abci::ConsensusResponse,
+            Error = tower::BoxError,
+        > + Send
+        + Clone
+        + 'static,
+    C::Future: Send + 'static,
+    C::Error: Sized,
+{
+    async fn fast_forward_to_next_epoch(
+        &mut self,
+        storage: &TempStorage,
+    ) -> Result<penumbra_sct::epoch::Epoch, anyhow::Error> {
+        use {penumbra_sct::component::clock::EpochRead, tap::Tap};
+
+        let get_epoch = || async { storage.latest_snapshot().get_current_epoch().await };
+        let start = get_epoch()
+            .await?
+            .tap(|start| tracing::info!(?start, "fast forwarding to next epoch"));
+
+        loop {
+            self.block().execute().await?;
```

Comment:
This completely works, and I think the helper method makes sense. For later, I think we will want to have a way to fast-forward to the next epoch as dynamically selected by the application. This requires only an extra look-up and gives us the capability to enshrine staking component invariant in tests (e.g. an easy one is that `Tombstoning` MUST trigger an epoch transition)

Reply:
> For later, I think we will want to have a way to fast-forward to the next epoch as dynamically selected by the application. 

x-ref #4149, where we can address this :slightly_smiling_face: